index.html

```diff
@@ -161,123 +144,134 @@ <h2>
-    </p>
-
-        <p>The <code>name</code> attribute will return the measure's name.</p>
-        <p>The <code>entryType</code> attribute will return the <a href='https://www.w3.org/TR/WebIDL/#idl-DOMString'><code>DOMString</code></a> <code>"measure"</code>.</p>
-        <p>The <code>startTime</code> attribute will return a <a href="https://w3c.github.io/hr-time/#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a>
-		with the measure's start mark [[!HR-TIME-2]].</p>
-        <p>The <code>duration</code> attribute will return a <a href="https://w3c.github.io/hr-time/#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a>
-		with the duration of the measure. </p>
+  <section>
+    <h2>Processing</h2>
+    <section>
+      <h2>Ensure <var>name</var> is valid</h2>
+      <p>To ensure the given mark or measure <var>name</var> is valid, run these steps:</p>
+      <ol>
+        <li>If the <a data-cite="!HTML51/webappapis.html#global-object">global object</a> is a <code>Window</code> object and <var>name</var> uses the same name as a <a data-cite="!WEBIDL#dfn-read-only">read only attribute</a> in the <code><a data-cite="!NAVIGATION-TIMING#performancetiming">PerformanceTiming</a></code> interface, <a data-cite="!WEBIDL#dfn-throw">throw</a> a <a data-cite="!WEBIDL#syntaxerror"><code>SyntaxError</code></a>.</li>
+      </ol>
+    </section>
+    <section>
+      <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
+      <p>To compute a <a data-cite="!HR-TIME-2#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a> given a <var>name</var> that is a <a data-cite="!WEBIDL#dfn-read-only">read only attribute</a> in the <code><a data-cite="!NAVIGATION-TIMING#performancetiming">PerformanceTiming</a></code> interface, run these steps:<p>
+      <ol>
+        <li>If the <a data-cite="!HTML51/webappapis.html#global-object">global object</a> is not a <code>Window</code> object, <a data-cite="!WEBIDL#dfn-throw">throw</a> a <a data-cite="!WEBIDL#syntaxerror"><code>SyntaxError</code></a>.</li>
+        <li>If <var>name</var> is <code>navigationStart</code>, return <code>0</code>.</li>
+        <li>Let <var>startTime</var> be the value of <code>navigationStart</code> in the <code><a data-cite="!NAVIGATION-TIMING#performancetiming">PerformanceTiming</a></code> interface.</li>
+        <li>Let <var>endTime</var> be the value of <var>name</var> in the <code><a data-cite="!NAVIGATION-TIMING#performancetiming">PerformanceTiming</a></code> interface.</li>
+        <li>If <var>endTime</var> is <code>0</code>, <a data-cite="!WEBIDL#dfn-throw">throw</a> an <a data-cite="!WEBIDL#invalidaccesserror"><code>InvalidAccessError</code></a>.</li>
+        <li>Return result of subtracting <var>startTime</var> from <var>endTime</var>.</li>
+      </ol>
+    </section>

```

Comment:
I think it would also be helpful to add a non-normative note, either as a preface, or after each algorithm here to note that: (a) PerformanceTiming was defined by NT1 and is now considered obsolete, (b) there are no plans to extend this functionality to PerformanceNavigationTiming (NT2).

Reply:
Done.